File: source/backend/python/application/domain/model/account.py
from money import Money

class Account:
    def __init__(self, account_id: str, balance: Money):
        self.account_id = account_id
        self.balance = balance

    def get_account_id(self):
        return  self.account_id

    def get_balance(self):
        return self.balance

    def deposit(self, money: Money) -> bool:
        """入金処理
        自身の口座に入金される
        Args:
            money(Money): 入金するお金クラス

        Returns:
            bool: 成功ならTrue, 失敗ならFalse
        """
        self.balance = Money.add(self.balance, money)
        return True

    def withdraw(self, money: Money) -> bool:
        """引き出し処理
        自身の口座からお金が引き出しされる
        Args:
            money(Money): 出金するお金クラス

        Returns:
            bool: 成功ならTrue, 失敗ならFalse
        """
        if self._may_withdraw(money=money) is False:
            return False

        self.balance = Money.subtract(self.balance, money)
        return True

    def _may_withdraw(self, money: Money) -> bool:
        """引出可能チェック
        口座から引き出した場合に残額が0未満にならないかを確認する
        Args:
            money(Money): お金クラス

        Returns:
            bool: 成功ならTrue, 失敗ならFalse
        """
        return Money.subtract(self.balance, money).is_positive()

File: source/backend/src/application/domain/model/Account.ts
/**
 * 口座に関するドメインモデル
 */

import { Money } from "./Money";

export class AccountId {
    public readonly id: string;

    constructor(id: string) {
        this.id = id;
    }
}

export class Account {
    private accountId: AccountId // 口座番号
    private balance: Money // 残高額

    constructor(accountId: AccountId, balance: Money) {
        this.accountId = accountId
        this.balance = balance
    }

    //  getter
    public getAccountId(): AccountId {
        return this.accountId;
    }

    public getBalance(): Money {
        return this.balance;
    }

    /**
     * 自身の口座に入金される
     * @param sourceAccountId 
     * @returns 
     */
    public deposit(sourceAccount: Account, money: Money): boolean {
        this.balance = Money.add(this.balance, money)
        return true;
    }

    /**
     * 自身の口座から引き出される
     * @returns 
     */
    public withdraw(targetAccount: Account, money: Money): boolean {
        if (!this.mayWithdraw(money)) return false;

        this.balance = Money.subtract(this.balance, money);
        return true;
    }

    /**
     * 引き出し可能かどうかをチェックする
     * @returns
     */
    private mayWithdraw(money: Money): boolean {
        return Money.subtract(this.balance, money).isPositive();
    }
}